src/transformers/integrations.py

```diff
@@ -1044,14 +1046,35 @@ def _initialize_run(self):
     @property
     def run(self):
         if self._run is None:
-            self._initialize_run()
+            self._reinitialize_run()
         return self._run
 
     def __del__(self):
         self._stop_run_if_exists()
 
+    @property
+    def _metadata_namespace(self):
+        return self.run[self._base_namespace_path]
+
+    def _log_integration_version(self):
+        self.run["source_code/integrations/transformers"] = version
+
+    def _log_trainer_parameters(self, args):
+        self._metadata_namespace["trainer_parameters"] = args.to_sanitized_dict()
+
+    def _log_model_parameters(self, model):
+        if model and hasattr(model, "config") and model.config is not None:
+            self._metadata_namespace["model_parameters"] = model.config.to_dict()
+
     def on_train_begin(self, args, state, control, model=None, **kwargs):
-        self._initialize_run()
+        if self._should_reinitialize:
+            self._reinitialize_run()
+        self._should_reinitialize = True
```

Comment:
I don't get this part:
* when `on_train_begin` may be called with `_should_reinitialize == false`?
* why do we have to reinitialize run after `on_train_begin`?

Reply:
Nevermind I've noticed this logic is changed in https://github.com/neptune-ai/transformers/pull/3/commits/aa2662a290be2a3229762b18f51c67af5acda99e

Reply:
Sorry :D. This logic will gets updates over and over :wink: